model/error/event.go

```diff
@@ -142,6 +141,10 @@ func DecodeEvent(input interface{}, err error) (transform.Transformable, error)
 		TransactionType:    decoder.StringPtr(raw, "type", "transaction"),
 	}
 
+	if cfg.Experimental {
```

Comment:
Did you consider pushing this decision to include `experimental` in the context all the way down to `DecodeContext`?  That would at least unify the decision making for transaction and errors but doesn't help for spans.